app/models/dialog.rb

```diff
@@ -47,7 +47,9 @@ def dialog_resources
   end
 
   def automate_values_hash
-    dialog_fields.each_with_object({}) { |df, result| result[df.automate_key_name] = df.automate_output_value }
+    dialog_fields.each_with_object({}) do |df, result|
+      result[MiqAeEngine.create_automation_attribute_array_key(df.automate_key_name)] = df.automate_output_value
```

Comment:
@d-m-u Not every attribute is going to be an array type should this be done only for array types?